examples/assembly/Add_Array_(AArch64).asm

```diff
@@ -0,0 +1,191 @@
+# corresponding to active lanes of p0 are stored into res.
+# Finally, the induction variable x4 is incremented by number of 32-bit words
+# in the vector.
+# If none of the predicate elements are active, the loop is terminated.
+#
+# Note that the code doesn't rely on knowing vector length, and can
+# thus "scale" to different vector lengths set by hardware without
+# needing recompilation, and also doesn't need a tail scalar loop.
+#
+# Eg: With vector length = 256 bits, ie 8 number of 32-bit elements.
+# 1st iteration:
+# p0.s = [x4 < x3, x4+1 < x3, ... x4+<len-1> < x3] = [ 1, 1, 1, 1, 1, 1, 1, 1]
+# Since first element of p0 is active, we branch to loop body,
+# Since all lanes of p0 are active, load 8 elements from A starting from &A[x4]
+# z0.s = [ A[0], A[1], A[2], A[3], A[4], A[5], A[6], A[7] ]
+# Similarly,
+# z1.s = [ B[0], B[1], B[2], B[3], B[4], B[5], B[6], B[7] ]
+# Add and comput result in z0:
+# z0.s = [ z0.s[0] + z0.s[1], ... +, z0.s[7] + z1.s[7] ] 
+# res = [ z0.s[0], z0.s[1], z0.s[2], z0.s[3], z0.s[4], z0.s[5], z0.s[6], z0.s[7] ]
+# Finally incw x4 will increment x4 by 8, thus x4 = 8.
+#
+# 2nd iteration:
+# x4 = 8
+# p0.s = [ 1, 1, 0, 0, 0, 0, 0, 0 ]
+# Since only first two lanes are active in p0, we load &A[8], &A[9]
+# in z0 and zero out rest of elements (because of /z): 
+# z0.s = [ A[8], A[9], 0, 0, 0, 0, 0, 0 ]
+# Similarly, z1.s = [ B[8], B[9], 0, 0, 0, 0, 0, 0 ]
+# Compute the result:
+# z0.s = [ z0.s[0] + z1.s[0], z0.s[1] + z1.s[1], 0, 0, 0, 0, 0, 0 ] 
+# and since first two elements of p0 are active:
+# we store z0.s[0] into res[8] and z0.s[1] in res[9] repsectively.
+# Increment x4 by 8, thus x4 = 16.
+#
+# 3rd iteration:
+# x4 = 16
+# Since x4 > N, p0 will be all false predicate.
+# Since all lanes of p0 are inactive, the loop is terminated.
+#
+# Same eg, with vector length = 512 bits, ie, 16 number of 32-bit elements.
+# 1st iteration:
+# p0.s = [ 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0 ]
+# z0.s = [ A[0], A[1], A[2], A[3], A[4], A[5], A[6], A[7],
+#	   A[8], A[9], 0, 0, 0, 0, 0, 0 ]
+# z1.s = [ B[0], B[1], B[2], B[3], B[4], B[5], B[6], B[7],
+#	   B[8], B[9], 0, 0, 0, 0, 0, 0 ]
+# Compute sum in z0:
+# z0.s = [ z0.s[0] + z1.s[0], ..., z0.s[9] + z1.s[9], 0, 0, 0, 0, 0, 0 ] 
+# Store first 9 elements from z0 since first 9 lanes of p0 are active.
+# res = [ z0.s[0], z0.s[1], z0.s[2], z0.s[3], z0.s[4], z0.s[5], z0.s[6],
+#	  z0.s[7], z0.s[8], z0.s[9] ]
+# Increment x4 by 16, thus x4 = 16.
+#
+# 2nd iteration:
+# p0.s = all false predicate since x4 > N.
+# Since all lanes of p0 are inactive, terminate the loop.
+
+addarrays_sve:
+    mov x4, xzr
+    b .cond_2
+.loop_body_2:
+    # Load elements in z0 corresponding to active lanes of p0
+    # and zero out the rest.
+    ld1w z0.s, p0/z, [x1, x4, lsl 2]
+    ld1w z1.s, p0/z, [x2, x4, lsl 2]
+    add z0.s, z0.s, z1.s
+    # Store elements of result computed in z0, corresponding to
+    # active lanes of p0 in x0 (res).
```

Comment:
"into x0 (res)."

Though I'm not sure what you mean by res.

Maybe:

"into x0 (which points to "res")."

Otherwise it looks like assembly syntax like `x0(res)` like you're offsetting x0 by res on MIPS.